### PR TITLE
Add `oauth` support to Admin Api calls.

### PR DESCRIPTION
**Pull request by nitzanj**

### CODE REVIEW

File: cloudinary-core/src/main/java/com/cloudinary/Configuration.java [hunk 1]
```diff
@@ -47,6 +47,7 @@
     public boolean forceVersion = true;
     public boolean longUrlSignature = DEFAULT_IS_LONG_SIGNATURE;
     public SignatureAlgorithm signatureAlgorithm = DEFAULT_SIGNATURE_ALGORITHM;
+    public String oauthToken = null;
```
Comment (const-cloudinary) [inline cloudinary-core/src/main/java/com/cloudinary/Configuration.java:50]: maybe call it oAuthToken ?

File: cloudinary-core/src/main/java/com/cloudinary/strategies/AbstractApiStrategy.java [hunk 2]
```diff
@@ -31,4 +32,19 @@ protected String createApiUrl (Iterable<String> uri, String prefix, String cloud
...
+        } else {
+            return "Basic " + Base64Coder.encodeString(apiKey + ":" + apiSecret);
+        }
+    }
+
+    protected void validateAuthorization(String apiKey, String apiSecret, String oauthToken) {
+        if (oauthToken == null) {
+            if (apiKey == null) throw new IllegalArgumentException("Must supply api_key");
+            if (apiSecret == null) throw new IllegalArgumentException("Must supply api_secret");
+        }
```
Comment (const-cloudinary) [inline cloudinary-core/src/main/java/com/cloudinary/strategies/AbstractApiStrategy.java:48]: what if all of them are null?

**Review by const-cloudinary — CHANGES_REQUESTED** [review]

File: cloudinary-core/src/main/java/com/cloudinary/Configuration.java (hunk 1 above)
Comment (nitzanj) [reply cloudinary-core/src/main/java/com/cloudinary/Configuration.java:50]: The camelCase name defined in the ticket is`oauth_token` so it kinda translates to oauthToken - but honestly one letter word in variables are always an issue. Not sure what makes more sense. @asisayag2 want to make the call?

File: cloudinary-core/src/main/java/com/cloudinary/strategies/AbstractApiStrategy.java (hunk 2 above)
Comment (nitzanj) [reply cloudinary-core/src/main/java/com/cloudinary/strategies/AbstractApiStrategy.java:48]: If all are null then this is thrown - `IllegalArgumentException("Must supply api_key"). Same behaviour as before for people not using oauth so it doesn't break anything.

**Review by const-cloudinary — APPROVED** [review]

File: cloudinary-core/src/main/java/com/cloudinary/Configuration.java (hunk 1 above)
Comment (asisayag2) [reply cloudinary-core/src/main/java/com/cloudinary/Configuration.java:50]: Looked at other Java OAuth libraries... 
`OAuthBlaBla` for objects
`oauthBlaBla` for members
So `oauthToken` it is.

File: cloudinary-core/src/main/java/com/cloudinary/Configuration.java [hunk 3]
```diff
@@ -87,11 +89,12 @@ private Configuration(
         this.proxyPort = proxyPort;
         this.secureCdnSubdomain = secureCdnSubdomain;
         this.useRootPath = useRootPath;
-        this.timeout = 0;
+        this.timeout = timeout;
```
Comment (asisayag2) [inline cloudinary-core/src/main/java/com/cloudinary/Configuration.java:92]: Why was that changed?

**Review by asisayag2 — APPROVED** [review]

File: cloudinary-core/src/main/java/com/cloudinary/Configuration.java (hunk 3 above)
Comment (nitzanj) [reply cloudinary-core/src/main/java/com/cloudinary/Configuration.java:92]: It's a bugfix (will be in the commit message on merge).
